pkg/checker/functools.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2017-2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package checker
+
+import (
+	"fmt"
+	"reflect"
+)
+
+func makeCheckRunner(f interface{}, args ...interface{}) (CheckRunner, error) {
+	if err := validateFunction(f, args...); err != nil {
+		return nil, fmt.Errorf("invalid function: %v", err)
+	}
+	argValues := make([]reflect.Value, 0, len(args))
+	for _, arg := range args {
+		argValues = append(argValues, reflect.ValueOf(arg))
+	}
+	return func() error {
+		ret := reflect.ValueOf(f).Call(argValues)
+		// len(ret) is assumed to be 1 and of type `error`, because of type
+		// `CheckRunner`
+		if ret[0].IsNil() {
+			return nil
+		}
+		return ret[0].Interface().(error)
```

Comment:
May be it worth to add a comment that this output signature is guaranteed by `validateFunction`

Reply:
done, also for `makeRemediationRunner`